core/src/main/scala/com/avast/clients/rabbitmq/converters.scala

```diff
@@ -35,7 +37,14 @@ trait ProductConverter[A] {
 object ProductConverter {
   implicit val identity: ProductConverter[Bytes] = new ProductConverter[Bytes] {
     override def convert(p: Bytes): Either[ConversionException, Bytes] = Right(p)
-
+    override def fillProperties(properties: MessageProperties): MessageProperties = properties
+  }
+  implicit val bytesArray: ProductConverter[Array[Byte]] = new ProductConverter[Array[Byte]] {
+    override def convert(p: Array[Byte]): Either[ConversionException, Bytes] = Right(Bytes.copyFrom(p))
+    override def fillProperties(properties: MessageProperties): MessageProperties = properties
+  }
+  implicit val utf8String: ProductConverter[String] = new ProductConverter[String] {
+    override def convert(p: String): Either[ConversionException, Bytes] = Right(Bytes.copyFromUtf8(p))
     override def fillProperties(properties: MessageProperties): MessageProperties = properties
```

Comment:
I've made it, and I've also made it for existing `bytes` converters.

Reply:
As I'm thinking about this now, I would prefer to remove these `properties` changes.

The problem is that `bytes` or `string` are actually _super-types_ for more content types. E.g. you want to generate a JSON and you cannot use circe so you will use the `utf8String` or `bytes` converter. And if you don't set the Content-Type explicitly then it will be automatically set to `text/plain` or `application/octet-stream`.

Then it's a big difference on the consumer side if there is no Content-Type or Content-Type set to `text/plain` or `application/octet-stream`.

Reply:
Yeah, I was aware that default (identity) converters didn't have this implemented too. However I'm not against, it's probably more correct.

Reply:
I've already removed the automatic `Content-Type` because I believe this behavior could be surprise for users...

Reply:
This has to be documented properly, I agree. However, I believe it's totally legitimate solution.